app/controllers/qa/linked_data_terms_controller.rb

```diff
@@ -0,0 +1,108 @@
+# This controller is used for all requests to linked data authorities. It will verify params and figure
+# out which linked data authority to query based on the 'vocab' param.
+
+class Qa::LinkedDataTermsController < ApplicationController
+  before_action :check_vocab_param, :init_authority
+  before_action :check_query_param, only: :search
+  before_action :check_id_param, only: :show
+
+  # Provide a warning if there is a request for all terms.
+  def index
+    logger.warn 'Linked data authorities do not support retrieving all terms.'
+    head :not_found
+  end
+
+  # Return a list of terms based on a query
+  def search
+    subauth = subauthority
```

Comment:
I'm not sure assigning to a local has any benefit here. There's no work being done in that method.